src/components/Form/Form.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import getApi from '../../services/api';
+import BodyDisplay from '../BodyDisplay/bodyDisplay';
+import HistoryList from '../HistoryList/historyList';
+
+export default class Form extends Component {
+
+  state = {
+    history: [],
+    apiResponses: [],
+    url: '',
+    method: '',
+    jsonEntry: ''
+  };
+
+  handleChange = ({ target }) => {
+    this.setState({ [target.name]: target.value });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    getApi(this.state.url, this.state.name)
+      .then(result => {
+        console.log(result);
+        this.setState({ apiResponses: result.results  });
+        this.setState(state => ({ history: [{ url: this.state.url, method: this.state.method }, ...state.history] }));
```

Comment:
just have one setState here. two in a row is no good
this.setState(state => ({ 
history: [
{ 
url: this.state.url, 
method: this.state.method 
},
 ...state.history
],
apiResponses: result.results
 }));